Create map.md, closes #3068

reganmarie opened the pull request:
<!--
👋 Hi, thanks for sending a PR to Codecademy Docs! 💖
Please fill out all fields below to ensure your PR is reviewed quickly.

**IMPORTANT**

If you would like to receive credit for your contribution to an entry, make sure to have your Codecademy user profile linked to your GitHub account:

1. Go to your Codecademy dashboard.
2. Click on your profile image in the top-right and then choose "Profile".
3. Then click "Edit Profile".
4. In the "GitHub Username" field, add your username (without the @).
5. Click "Save Changes"

Of course, you can opt not to do this and be listed as an "Anonymous contributor", instead. :)
-->

### Description

- Created map.md on filepath docs/content/go/concepts/strings/terms/map/map.md
- Added Title, Description, Subjects, Tags, and Catalog Content Added description, syntax, example, and codebyte example
- Ran through 3 online plagiarism checkers to ensure 100% unique wording and description, passed with 100%
- Followed content and style guidelines -Checked for spelling and grammar errors
- Closes issue #3068 

### Type of Change

<!--- Please delete or cross off the bullet point(s) that are irrelevant to this PR: -->

- Adding a new entry

### Checklist

<!-- Please check ALL the boxes: -->

- [X] All writings are my own.
- [X] My entry follows the Codecademy Docs style guide.
- [X] My changes generate no new warnings.
- [X] I have performed a self-review of my own writing and code.
- [X] I have checked my entry and corrected any misspellings.
- [X] I have made corresponding changes to the documentation if needed.
- [X] I have confirmed my changes are not being pushed from my forked `main` branch.
- [X] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [X] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

<!--
Having trouble with the PR checker? Here are some common issues and resolutions:

- verify_formatting is failing
  - run `yarn format path/to/markdown/file.md` or `yarn format:all` and commit the results
- verify_lint is failing
  - same as above
  - if verify_lint is still failing, running `yarn lint` locally should let you know what needs to be changed by hand
- test is failing
  - ensure any new markdown files have a `Title` and `Description` defined in their metadata
  - ensure any new markdown files only contain alphanumerics and dashes in their file names and have the same name as their parent directory
  - if that looks ok, running `yarn test` locally should let you know what the issue is
-->

caupolicandiaz commented on content/go/concepts/strings/terms/map/map.md:
```diff
@@ -0,0 +1,88 @@
+
+The `strings.Map()` function takes two parameters:
+
+- `funct(rune) rune`: The new character that the initial character will be replaced with.
```
```suggestion
- `mappingFunct(r)`: A function and a character (type rune) that the original character will be replaced with.
```

caupolicandiaz reviewed (changes requested):
@reganmarie I've completed my initial review. Please update the entry, and tag me when you've finished making changes. Thanks

reganmarie commented:
@caupolicandiaz Updated maps.md with your revisions! 
- Updated line 15 to "according" and added "given"
- Changed line 20 to "Map(mappingFunct(r),  s)"
- Changed line 25 to above syntax and added "A function and a character (type, rune)"
- Line 26 changed "string" to "s"
- Updated code blocks to include 2 spaces instead of tabs

caupolicandiaz reviewed (approved):
@reganmarie This is ready for a second review.

reganmarie commented:
Thank you @caupolicandiaz! ☺️

reganmarie commented:
@caupolicandiaz Made all of the changes requested! 
- Updated + and ( flag description in line 37
- Changed set_locale() to setlocale() in lines 70, 16, and 65. 
- Removed codebyte example since the compiler will likely no longer support this function soon.
- Ran lint to ensure proper formatting

reganmarie commented:
@caupolicandiaz So sorry about the comment - It was meant for the PHP money_format() pull request!

yangc95 commented on content/go/concepts/strings/terms/map/map.md:
```diff
@@ -0,0 +1,88 @@
+
+The `strings.Map()` function takes two parameters:
+
+- `mappingFunct(r)`: A function and a character (type rune) that the original character will be replaced with.
```
We don't have an entry on runes so it would be helpful here to making a differentiation here

yangc95 replied:
edit: I'm going to link the datatypes entry here

reganmarie replied:
Thank you for adding the link!

yangc95 commented on content/go/concepts/strings/terms/map/map.md:
```diff
@@ -0,0 +1,88 @@
+
+func main() {
+  mapped := func(r rune) rune {
+    if r == "e" {
```
I think you meant to use single quotes here for runes

```suggestion
    if r == 'e' {
```

yangc95 commented on content/go/concepts/strings/terms/map/map.md:
````diff
@@ -0,0 +1,88 @@
+The following example shows the implementation of `strings.Map()`:
+
+```go
+  package main
````
```suggestion
package main
```

yangc95 commented on content/go/concepts/strings/terms/map/map.md:
````diff
@@ -0,0 +1,88 @@
+## Syntax
+
+```pseudo
+Map(mappingFunct(r),  s)
````
```suggestion
strings.Map(mappingFunct(r), s)
```
